core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -275,7 +275,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     val clusterLinkTopic = createTopicsRequest.data.topics.stream().filter(t => t.clusterLink() != null && !t.clusterLink().isEmpty).findFirst()
     if (clusterLinkTopic.isPresent) {
       logger.info(s"!!! Handling create mirror topics request: ${clusterLinkTopic.get().clusterLink()}")
-      topicMirrorLinkCoordinator.addTopicsInCoordinator(clusterLinkTopic.get().clusterLink(), util.List.of(clusterLinkTopic.get().name()));
+      topicMirrorLinkCoordinator.addTopicsInCoordinator(clusterLinkTopic.get().clusterLink(), util.Set.of(clusterLinkTopic.get().name()));
```

Comment:
One thing that came to mind is that we don’t retrieve the remote topic configuration when creating the mirror topic. As a result, the topic configs and partition count will change after the TopicMirrorLinkCoordinator triggers refreshRemoteMetadata. Do you think we should fetch the remote topic config and partition count upfront, at the time of topic creation? Though I think this is a minor priority.

Reply:
Agree we should get the topic configs/partition count before creating the mirror topics. And I also agree it's a minor priority for PoC. We just need to prove we can achieve that with our current design, I think that should be fine.